pkg/sql/queuefeed/reader.go

```diff
@@ -166,20 +173,39 @@ func (r *Reader) setupRangefeed(ctx context.Context, assignment *Assignment) err
 			if checkpoint.ResolvedTS.IsEmpty() {
 				return
 			}
-			select {
-			case incomingResolveds <- checkpoint.ResolvedTS:
-			case <-ctx.Done():
-			default: // TODO: handle resolveds (dont actually default here)
+
+			r.mu.Lock()
+			defer r.mu.Unlock()
+
+			// Wait for rows to be read before adding more, if necessary.
+			for ctx.Err() == nil && len(r.mu.buf) > maxBufSize {
+				r.mu.poppedWakeup.Wait()
+			}
+
+			if ctx.Err() != nil {
+				return
 			}
+
+			r.mu.buf = append(r.mu.buf, bufferedEvent{resolved: checkpoint.ResolvedTS})
 		}),
 		rangefeed.WithOnInternalError(func(ctx context.Context, err error) { r.cancel(err) }),
 		rangefeed.WithConsumerID(42),
 		rangefeed.WithInvoker(func(fn func() error) error { return fn() }),
 		rangefeed.WithFiltering(false),
 	}
 
-	// TODO: resume from cursor
-	initialTS := hlc.Timestamp{WallTime: timeutil.Now().UnixNano()}
+	// Resume from checkpoint if available
+	// TODO: Support multiple partitions
+	partitionID := int64(1)
```

Comment:
we have a partition per range now. can you make it read/save them all? could be in another pr